cumulusci/cli/cci.py

```diff
@@ -809,13 +809,63 @@ def org_connect(config, org_name, sandbox, login_url, default, global_org):
     oauth_dict = oauth_capture()
     org_config = OrgConfig(oauth_dict, org_name)
     org_config.load_userinfo()
-
-    config.keychain.set_org(org_config, global_org)
+    org_config._load_orginfo()
+    datetime_format = "%Y-%m-%dT%H:%M:%S.%f+0000"
+    if org_config.organization_sobject["IsSandbox"]:
+        # confirmed org is scratch org
+        if org_config.organization_sobject["TrialExpirationDate"]:
+            days = (
+                datetime.strptime(
+                    org_config.organization_sobject["TrialExpirationDate"],
+                    datetime_format,
+                )
+                - datetime.strptime(
+                    org_config.organization_sobject["CreatedDate"], datetime_format
+                )
+            ).days
+            # need to handle this case!
+            if org_config.organization_sobject["NamespacePrefix"]:
+                org_config.config.update(
+                    {"namespaced": org_config.organization_sobject["NamespacePrefix"]}
+                )
+            org_config.config.update(
+                {
+                    "created": True,
+                    "config_file": "",
+                    "config_name": None,
+                    "email_address": org_config.userinfo__email,
+                    "org_id": org_config.userinfo__organization_id,
+                    "scratch_org_type": "workspace",
+                    # I think you need scratch value:
+                    # "scratch": True,
+                    "set_password": True,
+                    "connected_scratch": True,
+                    "sfdx_alias": "",
+                    "username": org_config.userinfo__preferred_username,
+                    "date_created": datetime.strptime(
+                        org_config.organization_sobject["CreatedDate"], datetime_format
+                    ),
+                    "days": days,
+                }
+            )
+            if not org_config.expired:
+                org_config.config.update({"active": True})
+        else:
+            # Unsure of this case below. I believe a sandbox that is
+            # not a scratch org is a trialforce org.
+            click.echo("Connecting as a trialforce org")
```

Comment:
Sandboxes don't have anything to do with Trialforce and they are a type of persistent org. You can do something like this for the overall structure of this code:

    is_scratch = org_config.organization_sobject["IsSandbox"] and org_config.organization_sobject["TrialExpirationDate"]
    if is_scratch:
        # connect as scratch org
    else:
        click.echo("Connecting as a persistent org.")